### PR TITLE
hop-node: L2Bridge: Use Type 0 tx type for Polygon commitTransfers txs

### PR DESCRIPTION
**Pull request by miguelmota**
- Uses `type: 0` for tx type for `commitTransfers` if source chain is polygon

### CODE REVIEW

**Comment by shanefontaine** [comment]
@miguelmota does the `type` need to be defined throughout the `GasBoostTransaction` class as well? Some poentital places: [1](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L108-L163), [2](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L238-L253) [3](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L261-L271), [4](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L300-L308) [5](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L589-L595)

**Comment by miguelmota** [comment]
@shanefontaine you are correct. The code has been updated and I added a test to verify type 0 txs

File: packages/hop-node/src/gasboost/GasBoostTransaction.ts [hunk 1]
```diff
@@ -374,7 +385,8 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
   }
 
   async getBumpedGasFeeData (multiplier: number = this.gasPriceMultiplier): Promise<Partial<GasFeeData>> {
-    const use1559 = await this.is1559Supported() && !this.gasPrice
+    const use1559 = await this.is1559Supported() && !this.gasPrice && this.type !== 0
```
Comment (shanefontaine) [inline packages/hop-node/src/gasboost/GasBoostTransaction.ts:388]: Technically if `use1559` should always be false if `tyis.type !== 0`, so maybe it is `const use1559 = (await this.is1559Supported() && !this.gasPrice) || this.type !== 0`

**Review by shanefontaine — APPROVED** [review]
@miguelmota one small comment, but feel free to merge after addressed. The rest LGTM :+1:

File: packages/hop-node/src/gasboost/GasBoostTransaction.ts (hunk 1 above)
Comment (shanefontaine) [reply packages/hop-node/src/gasboost/GasBoostTransaction.ts:388]: Updated the comment from `this.type !== 0` to `this.type ===0`

Comment (miguelmota) [reply packages/hop-node/src/gasboost/GasBoostTransaction.ts:388]: > Technically if use1559 should always be false if this.type === 0

That's what the `&& this.type !== 0` is doing

DeepinScreenshot_select-area_20220124130712

so I think it's correct

**Pull request merged**